### ISSUE TITLE
The local deployment of Qwen2.5-7B-Instruct for trial operation：APIConnectionError

### ISSUE THREAD

**Issue by Ensin1998** [OSU-NLP-Group/HippoRAG#121]
Hello, I modified some code with the help of GPT. When I attempted to use the locally deployed LLM API service, an APIConnectionError occurred, but I still got the final result. This should be wrong. I hope you can point out my problems or give me some suggestions for modifying the code. Thank you!

The trial run code is as follows:

`
from src.hipporag import HippoRAG
from src.hipporag.llm.local_qwen_llm import LocalQwenLLM
from src.hipporag.utils.config_utils import BaseConfig

config = BaseConfig()
config.openie_mode = 'online'
config.llm_name = 'Qwen2.5-7B-Instruct'
config.llm_base_url = "http://0.0.0.0:8000/v1/chat/completions"
config.embedding_model_name = '/home/yansen/model/nvidia/NV-Embed-v2'

hippo = HippoRAG(global_config=config)
hippo.llm_model = LocalQwenLLM(
    api_url=config.llm_base_url,
    model_name=config.llm_name,
    cache_enabled=True
)

queries = ["Where was Apple founded?"]
docs = ["Apple was founded by Steve Jobs in California."]
hippo.index(docs)
solutions, responses, metadata = hippo.rag_qa(queries)

print("Answer:", responses[0])
print("Metadata:", metadata[0])
`

The final output is as follows:
Loading checkpoint shards: 100%|████████████████████████████████████████████| 4/4 [00:02<00:00,  1.69it/s]
/home/yansen/.cache/huggingface/modules/transformers_modules/NV-Embed-v2/modeling_nvembed.py:349: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  'input_ids': torch.tensor(batch_dict.get('input_ids').to(batch_dict.get('input_ids')).long()),
/home/yansen/.conda/envs/hipporag/lib/python3.10/contextlib.py:103: FutureWarning: `torch.backends.cuda.sdp_kernel()` is deprecated. In the future, this context manager will be removed. Please see `torch.nn.attention.sdpa_kernel()` for the new context manager, with updated signature.
  self.gen = func(*args, **kwds)
NER: 100%|█| 1/1 [00:00<00:00, 7543.71it/s, total_prompt_tokens=174, total_completion_tokens=15, num_cache
Extracting triples: 100%|█| 1/1 [00:00<00:00, 948.29it/s, total_prompt_tokens=485, total_completion_tokens
1it [00:00, 30393.51it/s]
1it [00:00, 74898.29it/s]
KNN for Queries: 100%|██████████████████████████████████████████████████████| 1/1 [00:00<00:00, 18.87it/s]
100%|████████████████████████████████████████████████████████████████████| 3/3 [00:00<00:00, 25784.66it/s]
{'num_phrase_nodes': 3, 'num_passage_nodes': 1, 'num_total_nodes': 4, 'num_extracted_triples': 3, 'num_triples_with_passage_node': 3, 'num_synonymy_triples': 3, 'num_total_triples': 9}
Retrieving:   0%|                                                                   | 0/1 [00:00<?, ?it/sexception RetryError[<Future at 0x750bbaa54d00 state=finished raised APIConnectionError>]
/home/yansen/code/HippoRAG/src/hipporag/utils/misc_utils.py:105: RuntimeWarning: invalid value encountered in divide
  return (x - np.min(x)) / (np.max(x) - np.min(x))
Retrieving: 100%|███████████████████████████████████████████████████████████| 1/1 [00:03<00:00,  3.79s/it]
Collecting QA prompts: 100%|█████████████████████████████████████████████| 1/1 [00:00<00:00, 12157.40it/s]
QA Reading: 100%|████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 12192.74it/s]
Extraction Answers from LLM Response: 1it [00:00, 46091.25it/s]
Answer: The passage states that "Apple was founded by Steve Jobs in California." Therefore, the location where Apple was founded is California.
Answer: California.
Metadata: {'finish_reason': 'stop', 'prompt_tokens': 761, 'completion_tokens': 30}

**Comment by bernaljg** [comment]
We cache the LLM output to speed things up. Did you run this same exact script with the LLM API working by any chance?

**Comment by Ensin1998** [comment]
Thank you for your reminder. I'll check the code again carefully
